src/components/update-service/update-service.controller.ts

```diff
@@ -213,74 +213,28 @@ export class UpdateServiceController implements angular.IController {
     this.ctrl.orderComplete = false;
     this.ctrl.error = null;
 
-    var parameterData: any = this.getParameters(this.ctrl.parameterData);
+    let parameterData: any = this.getParameters(this.ctrl.parameterData);
 
     // Update the instance parameters
-    var originalParameters: any = _.get(this.ctrl.serviceInstance, 'spec.parameters');
-    var originalKeys: any = _.map(originalParameters, (value: any, key: string) => {
-      // Make sure properties with a '.' in the name aren't treated as paths.
+    let originalParameters: object = _.get(this.ctrl.serviceInstance, 'spec.parameters');
+    // NOTE: should be string[][], but compiler doesn't handle this
+    let originalKeys: any[] = _.map(originalParameters, (value: any, key: string) => {
       return [key];
     });
-    var updateParameters: any = _.pick(parameterData, originalKeys);
-    var newParameters: any = _.omit(parameterData, originalKeys);
-    var updateInstance: any = angular.copy(this.ctrl.serviceInstance);
+    let updateParameters: object = _.pick(parameterData, originalKeys);
+    let newParameters: object = _.omit(parameterData, originalKeys);
+    let updateInstance: object = angular.copy(this.ctrl.serviceInstance);
```

Comment:
Not sure object vs any buys us much... :/

I'd probably just use

```ts
const updateParameters = _.pick(parameterData, originalKeys);
const newParameters = _.omit(parameterData, originalKeys);
const updateInstance = angular.copy(this.ctrl.serviceInstance);
```

Reply:
Not much, but is aiming in the right direction I think :/